### PR TITLE
[BWA-86] Debug Menu #4

### PR DESCRIPTION
**Pull request by andrebispo5**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/BWA-86

## 📔 Objective

Bring over the debug menu files from the main app and adjust the navigation flows.
## 📸 Screenshots

https://github.com/user-attachments/assets/98259903-f8dd-48c2-95f5-e058b1b023f1

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/kotlin/com/bitwarden/authenticator/MainViewModel.kt [hunk 1]
```diff
@@ -37,16 +40,24 @@ class MainViewModel @Inject constructor(
...
         when (action) {
             is MainAction.Internal.ThemeUpdate -> handleThemeUpdated(action)
             is MainAction.ReceiveFirstIntent -> handleFirstIntentReceived(action)
             is MainAction.ReceiveNewIntent -> handleNewIntentReceived(action)
+            MainAction.OpenDebugMenu -> handleOpenDebugMenu()
         }
     }
 
+    private fun handleOpenDebugMenu() {
+        sendEvent(MainEvent.NavigateToDebugMenu)
```
Comment (SaintPatrck) [inline app/src/main/kotlin/com/bitwarden/authenticator/MainViewModel.kt:58]: Can we get a test for this event?

File: app/src/main/kotlin/com/bitwarden/authenticator/data/platform/repository/DebugMenuRepositoryImpl.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import com.bitwarden.authenticator.data.platform.manager.getFlagValueOrDefault
+import com.bitwarden.authenticator.data.platform.manager.model.FlagKey
+import com.bitwarden.authenticator.data.platform.repository.util.bufferedMutableSharedFlow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.onSubscription
+
+/**
+ * Default implementation of the [DebugMenuRepository]
+ */
+class DebugMenuRepositoryImpl(
```
Comment (SaintPatrck) [inline app/src/main/kotlin/com/bitwarden/authenticator/data/platform/repository/DebugMenuRepositoryImpl.kt:14]: Can we get tests for this repo?

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/debugmenu/DebugMenuScreen.kt [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+import com.bitwarden.authenticator.ui.platform.feature.debugmenu.components.ListItemContent
+import com.bitwarden.authenticator.ui.platform.theme.AuthenticatorTheme
+
+/**
+ * Top level screen for the debug menu.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Suppress("LongMethod")
+@Composable
+fun DebugMenuScreen(
```
Comment (SaintPatrck) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/debugmenu/DebugMenuScreen.kt:43]: Can we get tests for this screen?

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/debugmenu/DebugMenuViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+/**
+ * ViewModel for the [DebugMenuScreen]
+ */
+@HiltViewModel
+class DebugMenuViewModel @Inject constructor(
```
Comment (SaintPatrck) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/debugmenu/DebugMenuViewModel.kt:22]: Can we get tests for this view model?

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/debugmenu/components/FeatureFlagListItems.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import com.bitwarden.authenticator.R
+import com.bitwarden.authenticator.data.platform.manager.model.FlagKey
+import com.bitwarden.authenticator.ui.platform.components.toggle.BitwardenWideSwitch
+
+/**
+ * Creates a list item for a [FlagKey].
+ */
+@Suppress("UNCHECKED_CAST")
+@Composable
+fun <T : Any> FlagKey<T>.ListItemContent(
```
Comment (SaintPatrck) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/debugmenu/components/FeatureFlagListItems.kt:15]: Tests for DebugMenuScreen should give us coverage of this extension.

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/debugmenu/manager/DebugLaunchManagerImpl.kt [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import android.view.MotionEvent
+import com.bitwarden.authenticator.data.platform.repository.DebugMenuRepository
+
+private const val TAP_TIME_THRESHOLD_MILLIS = 500
+private const val POINTERS_REQUIRED = 3
+
+/**
+ * Default implementation of the [DebugMenuLaunchManager]
+ */
+class DebugLaunchManagerImpl(
```
Comment (SaintPatrck) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/debugmenu/manager/DebugLaunchManagerImpl.kt:14]: Missing tests `DebugLaunchManagerImpl`.

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/rootnav/RootNavScreen.kt [hunk 7]
```diff
@@ -80,6 +81,11 @@ fun RootNavScreen(
                 viewModel.trySendAction(RootNavAction.Internal.AppUnlocked)
             },
         )
+        setupDebugMenuDestination(
+            onNavigateBack = {
+                navController.popBackStack()
+            },
+        )
```
Comment (SaintPatrck) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/platform/feature/rootnav/RootNavScreen.kt:88]: Can we cover these lines?

**Comment by andrebispo5** [comment]
Sorry @SaintPatrck for some reason I totally missed the test files in the main app, I blame the IDE 😆

**Review by SaintPatrck — APPROVED** [review]

**Pull request merged**